### ISSUE TITLE
S2589 FP: Rule is raised on catch-all tuples in switch expressions

### ISSUE THREAD

**Issue by rjgotten** [SonarSource/sonar-dotnet#8008]
### Description

S2589 is raised on catch-all tuples in switch expressions.
This is a common pattern when using truth-table like exhaustive switch expressions when aligning tuple members for a consistent condition style. (See the repro steps.)

### Repro steps

```cs
class Person
{
  public int? Age { get; set; }
}

// ...

public int Compare(Person x, Person y) => (x.Age, y.Age) switch
{
  (null, null) =>  0,
  (null,    _) =>  1,
  (   _, null) => -1,
  (   _,    _) => Comparer<int>.Default.Compare(x.Age.Value, y.Age.Value)
};
```

### Expected behavior

S2589 is not raised.

### Actual behavior

S2589 is raised on the `(_, _)` case.

### Known workarounds

Use a plain catch-all `_` case instead - but this breaks the stylistic similarity and alignment of all the switch expression cases.

### Related information

* C#/VB.NET Plugins version: 9.8 (via SonarLint v7.3.0.77872)
* Visual Studio version: VS2022 v17.7.4
* MSBuild / dotnet version: .NET SDK 7.0.401
* SonarScanner for .NET version (if used): N/A
* Operating System: Windows 11

**Comment by gregory-paidis-sonarsource** [comment]
Hey @rjgotten ,

Thanks for the information.
I added a reproducer you can find here: #8014 

I also added to our dedicated backlog, where it will be taken care of sometimes in the future.

**Comment by mary-georgiou-sonarsource** [comment]
Another report related to the tuples: https://community.sonarsource.com/t/switch-expression-s2589-false-postive-or-documentation-gap/99560
